src/header-utilities/exponentialSearch.hpp

```diff
@@ -0,0 +1,51 @@
+class ExponentialSearch {
+  template< typename RandomAccessIterator,
+	    typename Value,
+	    typename Compare >
+  static RandomAccessIterator apply( RandomAccessIterator left,
+				     RandomAccessIterator right,
+				     RandomAccessIterator end,
+				     Value&& value,
+				     Compare&& compare ){
```

Comment:
I like the idea of adopting conventions from the standard library (in general). But we need to somehow communicate that to the end user. Not exactly sure how to do that as we don't have a great documentation system right now.